lib/spack/spack/cmd/versions.py

```diff
@@ -68,7 +65,7 @@ def versions(parser, args):
         if args.safe:
             return
 
-    fetched_versions = pkg.fetch_remote_versions(args.concurrency)
+    fetched_versions = pkg.fetch_remote_versions(args.jobs)
```

Comment:
This is likely the cause of the regression you see. For `versions` maybe it's better to oversubscribe up to a certain point.